pkg/manifests/manifests.go

```diff
@@ -1856,6 +1860,36 @@ func (f *Factory) setupQueryLogFile(p *monv1.Prometheus, queryLogFile string) er
 	return nil
 }
 
+func (f *Factory) setupScrapeProfiles(p *monv1.Prometheus, scrapeProfile string) error {
+	if scrapeProfile == "" {
+		return nil
+	}
+
+	if scrapeProfile != "full" && scrapeProfile != "operational" && scrapeProfile != "uponly" {
+		return errors.Wrap(ErrConfigValidation, `scrape profile provided is unknown, supported scrape profiles are: full, operational, uponly`)
+	}
+
+	labelSelector := &metav1.LabelSelector{
+		MatchExpressions: []metav1.LabelSelectorRequirement{
```

Comment:
@simonpasquier I was hoping this approach would work however the selectors are ANDed, so in this case, when we configure a scrape profile, Prometheus ends up not scraping anything... The complex problem here is joining two different sets of Generators (ServiceMonitor, PodMonitor and Probe), one with no profile label configured and one with a profile label configured. After thinking about it, I came up with 2 options:
1. Create another Prometheus CRD to select Generators based on the `profile` label. By Default we would start running 2 Prometheus CRDs one without selectors and another with `profile: full`. The problem with this solution is that it goes against the goal of the issue which is to lower resource consumption.
2. Introduce a change to Prometheus operator that would allow the Prometheus CRD to be configured with an array of `labelSelectors` that would be ORed. The problem with this solution is that we would be changing the Prometheus CRD API.

Reply:
**UPDATE** After some more testing, if we use the key selector `NotIn`  we are able to get the behaviour we want, so please disregard my comment